analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContextWrapper.cs

```diff
@@ -165,9 +164,9 @@ static SymbolStartAnalysisContextWrapper()
         {
             RegisterSyntaxNodeActionCS(RoslynSymbolStartAnalysisContext, action, syntaxKinds.Cast<CS.SyntaxKind>().ToImmutableArray());
         }
-        else if (languageKindType == typeof(VB.SyntaxKind))
+        else if (languageKindType.FullName == "Microsoft.CodeAnalysis.VisualBasic.SyntaxKind")
```

Comment:
can we remove this? is this necessary?

Reply:
No. We need to keep the check. It prevents the `throw` in the else path below.